pkg/theme/text.go

```diff
@@ -0,0 +1,395 @@
+	case RGBTextStyle:
+		rgbTextStyle := b.convertToRGB()
+
+		for _, opt := range other.GetOpts() {
+			rgbTextStyle.setOpt(opt, true)
+		}
+		rgbTextStyle.fg = typedOther.fg
+		rgbTextStyle.style.SetFg(typedOther.fg)
+
+		if typedOther.bg != nil {
+			// Making sure to copy value
+			rgbTextStyle.bg = &color.RGBColor{}
+			*rgbTextStyle.bg = *typedOther.bg
+			rgbTextStyle.style.SetBg(*typedOther.bg)
+		}
+
+		return rgbTextStyle
+	default:
+		return b
+	}
+}
+
+type RGBTextStyle struct {
+	opts  color.Opts
+	fg    color.RGBColor
+	bg    *color.RGBColor
+	style color.RGBStyle
+}
+
+func (b RGBTextStyle) Sprint(a ...interface{}) string {
+	return b.style.Sprint(a...)
+}
+
+func (b RGBTextStyle) Sprintf(format string, a ...interface{}) string {
+	return b.style.Sprintf(format, a...)
+}
+
+func (b RGBTextStyle) setOpt(opt color.Color, v bool) RGBTextStyle {
+	if v {
+		// Add value
+		for _, listOpt := range b.opts {
+			if listOpt == opt {
+				return b
+			}
+		}
+		b.opts = append(b.opts, opt)
+	} else {
+		// Remove value
+		for idx, listOpt := range b.opts {
+			if listOpt == opt {
+				b.opts = append(b.opts[:idx], b.opts[idx+1:]...)
+				return b
+			}
+		}
+	}
+	return b
+}
+
+func (b RGBTextStyle) SetBold(v bool) TextStyle {
+	b = b.setOpt(color.OpBold, v)
+	b.style.SetOpts(b.opts)
+	return b
+}
+
+func (b RGBTextStyle) SetReverse(v bool) TextStyle {
+	b = b.setOpt(color.OpReverse, v)
+	b.style.SetOpts(b.opts)
+	return b
+}
+
+func (b RGBTextStyle) SetUnderline(v bool) TextStyle {
+	b = b.setOpt(color.OpUnderscore, v)
+	b.style.SetOpts(b.opts)
+	return b
+}
+
+func (b RGBTextStyle) SetColor(v color.Color, background bool) TextStyle {
+	rgbColor := v.RGB()
+	if background {
+		b.bg = &rgbColor
+		b.style.SetBg(rgbColor)
+	} else {
+		b.fg = rgbColor
+		b.style.SetFg(rgbColor)
+	}
+	return b
+}
+
+func (b RGBTextStyle) SetRGBColor(red, green, blue uint8, background bool) TextStyle {
+	parsedColor := color.Rgb(red, green, blue, background)
+	if background {
+		b.bg = &parsedColor
+		b.style.SetBg(parsedColor)
+	} else {
+		b.fg = parsedColor
+		b.style.SetFg(parsedColor)
+	}
+	return b
+}
+
+func (b RGBTextStyle) GetOpts() color.Opts {
+	return b.opts
+}
+
+// Merge other on top of the text style
+func (b RGBTextStyle) Merge(other TextStyle) TextStyle {
+	var otherRgb RGBTextStyle
+
+	switch typedOther := other.(type) {
+	case BasicTextStyle:
+		otherRgb = typedOther.convertToRGB()
+	case RGBTextStyle:
+		otherRgb = typedOther
+	default:
+		return b
+	}
+
+	for _, opt := range b.GetOpts() {
+		switch opt {
+		case color.OpBold:
+			b.opts[0] = opt
+		case color.OpReverse:
+			b.opts[1] = opt
+		case color.OpUnderscore:
+			b.opts[2] = opt
+		}
+	}
+
+	b.fg = otherRgb.fg
+	b.style.SetFg(otherRgb.fg)
+
+	if otherRgb.bg != nil {
+		// Making sure to copy value
+		b.bg = &color.RGBColor{}
+		*b.bg = *otherRgb.bg
+		b.style.SetBg(*otherRgb.bg)
+	}
+
+	return b
+}
+
+func SetConfigKeys(s TextStyle, keys []string, background bool) TextStyle {
```

Comment:
I think SetConfigStyles might be a clearer name because SetConfigKeys might make the reader think of keybindings

Reply:
Fixed